server/src/main/java/io/spine/system/server/SystemFeatures.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * Enables the configured system context to store
+     * {@link io.spine.system.server.CommandLog CommandLog}s for domain commands.
+     *
+     * @return self for method chaining
+     * @see #disableCommandLog()
+     */
+    @CanIgnoreReturnValue
+    public SystemFeatures enableCommandLog() {
+        this.commandLog = true;
+        return this;
+    }
+
+    /**
+     * Disables {@linkplain io.spine.system.server.CommandLog CommandLog}.
+     *
+     * <p>This is the default setting.
+     *
+     * @return self for method chaining
+     * @see #enableCommandLog()
+     */
+    @CanIgnoreReturnValue
+    public SystemFeatures disableCommandLog() {
+        this.commandLog = false;
+        return this;
+    }
+
+    /**
+     * Enables querying of the latest domain {@code Aggregate} states.
+     *
+     * <p>The system context stores domain {@code Aggregate} states in the form of
+     * {@link io.spine.system.server.Mirror} projections.
+     *
+     * <p>This is the default setting.
+     *
+     * @return self for method chaining
+     * @see #disableAggregateQuerying()
+     */
+    @CanIgnoreReturnValue
+    public SystemFeatures enableAggregateQuerying() {
+        this.aggregateMirrors = true;
+        return this;
+    }
+
+    /**
+     * Disables querying of the latest domain {@code Aggregate} states.
+     *
+     * @return self for method chaining
+     * @see #enableAggregateQuerying()
+     */
+    @CanIgnoreReturnValue
+    public SystemFeatures disableAggregateQuerying() {
+        this.aggregateMirrors = false;
+        return this;
+    }
+
+    /**
+     * Configures the the system context to store system events.
+     *
+     * @return self for method chaining
+     */
+    public SystemFeatures persistEvents() {
+        this.storeEvents = true;
+        return this;
+    }
+
+    /**
+     * Configures the the system context NOT to store system events for better performance.
+     *
+     * <p>This is the default setting.
+     *
+     * @return self for method chaining
+     */
+    public SystemFeatures forgetEvents() {
+        this.storeEvents = false;
+        return this;
+    }
+
+    /**
+     * Obtains the {@link io.spine.system.server.CommandLog CommandLog} setting.
+     *
+     * @return {@code true} if the {@code CommandLog} should be stored, {@code false} otherwise
+     */
+    boolean includeCommandLog() {
+        return commandLog;
+    }
+
+    /**
+     * Obtains the {@code Aggregate} mirrors setting.
+     *
+     * @return {@code true} if the Aggregate mirrors should be stored, {@code false} otherwise
+     */
+    boolean includeAggregateMirroring() {
+        return aggregateMirrors;
+    }
+
+    /**
+     * Obtains the system events persistence setting.
+     *
+     * @return {@code true} if system events should be stored, {@code false} otherwise
+     */
+    @Internal
+    public boolean includePersistentEvents() {
+        return storeEvents;
+    }
+}
```

Comment:
`equals()`, `hashCode()`, `toString()`?

Reply:
On the other note, if we were building a server environment in another language, could it be a `Message`-based type? Don't we describe a `System` context after all?